playground/webpack/parseMarkdown.js

```diff
@@ -16,9 +27,65 @@ module.exports = function loader() {
   });
   this.cacheable();
 
-  const stringyData = JSON.stringify(parseMarkdown(files), null, 2);
+  const data = parseMarkdown(files);
+
+  // Work around JSON.stringify() not supporting functions.
+  // First replace all code functions within the data with a placeholder string.
+  // This transforms:
+  // { code: function() {/* blah */ } }
+  // into:
+  // { code: "___CODEPLACEHOLDER__0__0___" }
+  const dataWithPlaceholders = data.map((readme, readmeIdx) => ({
+    ...readme,
+    examples: readme.examples.map((example, exampleIdx) => ({
+      ...example,
+      code: `___CODEPLACEHOLDER__${readmeIdx}__${exampleIdx}___`,
+    })),
+  }));
+
+  // Then stringify the data, and replace all the placeholder strings with the
+  // with the function declaration.
+  // This transforms:
+  // { code: "___CODEPLACEHOLDER__0__0___" }
+  // back into:
+  // { code: function() {/* blah */ } }
+  const stringyData = JSON.stringify(dataWithPlaceholders, null, 2).replace(
+    /"___CODEPLACEHOLDER__(\d+)__(\d+)___"/g,
+    (_, readmeIdx, exampleIdx) =>
+      data[readmeIdx].examples[exampleIdx].code.toString(),
+  );
+
+  // Example code does not have any scope attached to it by default. It boldly
+  // states `<Button>An example Button</Button>`, blindly trusting that `Button`
+  // is available in its scope.
+  //
+  // codeInvoker is responsible for providing a scope for an example function
+  // so that it will work. It does this by creating a new fuction with the scope
+  // defined as parameters and then calling that new function.
+  // Assuming it is called with
+  // codeInvoker(function() { return (<Button>Hi</Button>) } {React, Button})
+  // It will transform:
+  // function() { return (<Button>Hi</Button>) }
+  // into:
+  // function(React, Button) { return (<Button>Hi</Button>) }
+  // and then call that function with React and Button as the arguments
+  const codeInvoker = function(fn, scope) {
+    const scopeKeys = Object.keys(scope);
+    const scopeValues = scopeKeys.map((key) => scope[key]);
+
+    // Replace the empty parameter list with a list based upon the scope.
```

Comment:
This is just my opinion so if you disagree feel free to leave it as is:

I really don't like long explanatory comments for every line like this. I usually feel like they make the code less readable by breaking up the code block. I'd prefer to have this included in the explanation above.